coral-hive/src/test/java/com/linkedin/coral/hive/hive2rel/CoralTestUDTF.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2018-2021 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.hive.hive2rel;
+
+import java.util.ArrayList;
+
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDTF;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspectorFactory;
+import org.apache.hadoop.hive.serde2.objectinspector.StructObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory;
+
+// This is used in TestUtils to set up as dali function
+// This needs in a separate file for Hive to correctly load for setup
+
+
+/**
+ * CoralTestUDTF outputs the number of rows seen, twice. It's output twice
+ * to test outputting of rows on close with lateral view.
+ */
+public class CoralTestUDTF extends GenericUDTF {
```

Comment:
Not specific to this PR. Do you think if there is way to reuse the UDF/UDTF class in other modules instead of duplicating the code?

Reply:
At present, I have no idea about how to reuse these classes since the tests are independent, and there will be exceptions if we don't create the UDF classes for each module's tests.

Reply:
Fair.